### PR TITLE
Add resistor-color-trio exercise

### PR DESCRIPTION
**Pull request by glaxxie**
First timer 🙏

### CODE REVIEW

**Review by BNAndras — COMMENTED** [review]
Looks good to me. @pfertyk, at some point we should sort the exercises by difficulty group as shown on the website (I believe it's 1-3 for easy, 4-6 for medium, 7-10 for hard) and then by exercise name within each group. The reason I mention this is that the resistor-color series is a notable exception where it'd be better to sort them sequentially. That lets the student see how each step leads into the next one from resistor-color to resistor-color-trio

**Comment by BNAndras** [comment]
Another thing to float. Do we want the inputs to be arrays of colors or do we want to require three arguments, one for each color. The latter would make the last test about ignoring additional colors unnecessary. All the other tests use three colors.

At the moment, the ported resistor-color-duo uses an array of colors so we should be consistent, but that doesn't preclude us from changing resistor-color-duo to use two arguments, one for each color.

File: exercises/practice/resistor-color-trio/.meta/example.gd [hunk 1]
```diff
@@ -0,0 +1,18 @@
+func color_code(colors):
+    var colors_code = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
```
Comment (pfertyk) [inline exercises/practice/resistor-color-trio/.meta/example.gd:2]: Could you please change space indentation to tab indentation? I promise that at some point I will write a small CI script to detect spaces, but for now we need to check this manually :)

GDScript will work with both spaces and tabs, as long as they are used consistently. However, the main programming environment of GDScript is Godot Editor, and it uses tabs by default.

**Comment by pfertyk** [comment]
@BNAndras noted, I already try to sort the exercises by difficulty, but we can update the whole list later :+1: 

As to arrays vs separate arguments, I am OK with both solutions. I would, however, lean a bit toward an array, as it will be consistent with the Python track. GDScript is very similar to Python, and I like to keep the solutions similar to make things easier for people who already completed the exercises in Python.

On the other hand, separate arguments will be "consistent" with the first exercise of this series (which accepts a single argument, not a single element array) :P What are your thoughts @glaxxie ?

**Comment by BNAndras** [comment]
Yeah, I wouldn’t have thought of it but the Julia maintainers mentioned it when I ported resistor-color-duo. They still ended up using an array because resistor-color-trio was already ported that way. I think the array approach is pretty typical from what I’ve seen in other tracks.

**Comment by glaxxie** [comment]
Fixed the spacing issue. I didn't use the godot editor for it since I was fiddling about for far too long, so I used vscode for the first PR.
The tab there is configured to be 4 spaces, was a bit surprised to see actual tab is 8 spaces on github.

Regarding the point about using array vs args: I would favor array as well.

I actually did have a conversation related this when code review this exercise with a learner. They asked why in the `resistor expert` version they should return '0 ohms' early if there is only one color in the array, and what if it was green. I dig around and the only instance of a resistor having one band I can found is the black one. 

From that, we can rationalize that the very first exercise is more about introducing the idea of assigning a color to retrieve a certain  value so using one arg (string) here is ok, and not really an accurate portrait of how resistor should be like later ones (array of colors), just like there isn't one with a leading black color (0) in later exercises.

**Comment by BNAndras** [comment]
Yeah, that GitHub thing led to a few issues for me on the first few PRs until I got the hang of it. :)

File: exercises/practice/resistor-color-trio/.meta/example.gd [hunk 2]
```diff
@@ -0,0 +1,18 @@
+func color_code(colors):
+	var colors_code = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
+
+	var units = {
+        1e9: "gigaohms",
```
Comment (pfertyk) [inline exercises/practice/resistor-color-trio/.meta/example.gd:5]: Much to my surprise, this file is correct by Godot Engine's standards, despite mixing spaces and tabs ... You learn something every new every day I guess ;)

Nevertheless, could you please replace spaces with tabs here as well? Just to keep things consistent ;) The rest of the PR looks good, once this is changed I will merge it :+1:

**Comment by pfertyk** [comment]
Thank you for contributing @glaxxie ! :)

**Pull request merged**